src/webview/panel-views/abstractPanelViewProxy.ts

```diff
@@ -0,0 +1,139 @@
+import { PanelViewProxy } from './panelViewProxy';
+import { ExtensionContext, Uri, Webview } from 'vscode';
+import { PanelViewColors, PanelViewInputVariableMap, PanelViewVariable } from '../../model/panelViewInput';
+import { readFileSync } from 'fs';
+import { isEqual } from 'lodash';
+import {
+  ExportImagePanelViewProxyMessage,
+  InitializeRenderingAreaPanelViewProxyMessage,
+  StartGifRecordingPanelViewProxyMessage,
+  StartWebMRecordingPanelViewProxyMessage,
+  StopGifRecordingPanelViewProxyMessage,
+  StopWebMRecordingPanelViewProxyMessage,
+  UpdateRenderingAreaPanelViewProxyMessage,
+} from './PanelViewProxyMessage';
+import { NodeModulesKeys } from '../../node-modules-accessor/nodeModulesKeys';
+import { NodeModulesAccessor } from '../../node-modules-accessor/nodeModulesAccessor';
+
+interface PathReplacement {
+  tag: string;
+  uri: Uri;
+}
+
+class PathHelper {
+  private readonly extensionUri: Uri;
+
+  private static getFileName(fileNameStub: string, extension: string): string {
+    return `${fileNameStub}.${extension}`;
+  }
+
+  constructor(private readonly webview: Webview, { extensionUri }: ExtensionContext) {
+    this.extensionUri = extensionUri;
+  }
+
+  createMediaReplacement(extension: string, fileNameStub: string): PathReplacement {
+    return {
+      tag: PathHelper.getFileName(fileNameStub, extension),
+      uri: this.getMediaUri(extension, fileNameStub),
+    };
+  }
+
+  getMediaUri(extension: string, fileNameStub: string): Uri {
+    return this.getUri('media', extension, PathHelper.getFileName(fileNameStub, extension));
+  }
+
+  getUri(...pathSegments: string[]): Uri {
+    return this.webview.asWebviewUri(Uri.joinPath(this.extensionUri, ...pathSegments));
+  }
+}
```

Comment:
Export this class to a separate file in the `util` directory.

Reply:
Done. I moved `PathReplacement` as well.